### PR TITLE
chore: docker compose to standalone by default

### PR DESCRIPTION
**Pull request by huangjw806**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

After testing, the standalone mode has become relatively stable, so we plan to change docker-compose to standalone by default.

1. `docker-compose.yml` -> `docker-compose-minio.yml`
2. `docker-compose-standalone.yml` -> `docker-compose.yml`

3. change integration tests docker-compose to standalone mode. The following two cases cannot be simply changed to standalone mode, so I will still deploy using a distributed cluster.
 - doris-sink 
 - feature-store 
 @xxhZs please help confirm if these two cases can be run in standalone mode.

https://github.com/risingwavelabs/risingwave/issues/13235
## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxhZs** [comment]
"Doris Sink" is because we need to ensure that the entire cluster adopts the same set of network IP, otherwise it will result in RW being unable to access Doris. Is there any other way to solve the whole problem? The Feature Store needs to transfer UDF files to Docker, and I'm not sure if there is any other solution.

File: docker/README.md [hunk 1]
```diff
@@ -45,12 +45,20 @@ To start a RisingWave playground, run
...
+docker-compose up
+```
+
 ### minio
 To start a RisingWave cluster with minio backend, run
 
 ```
 # Start all components
-docker-compose up
+docker-compose -f docker-compose-minio.yml up
```
Comment (kwannoel) [inline docker/README.md:61]:
```suggestion
docker-compose -f docker-compose-distributed.yml up
```

**Review by kwannoel — APPROVED** [review]
LGTM, let's trigger integration test on this PR before merging.
Also `docker-compose-minio` seems confusing since standalone mode has that too. We can just call it `docker-compose-distributed` instead.

**Comment by huangjw806** [comment]
standalone mode integration test：
https://buildkite.com/risingwavelabs/integration-tests/builds/488

It seems that the data check of vector case fails in standalone mode. Please help to check. cc @neverchanje

File: integration_tests/doris-sink/docker-compose.yml [hunk 2]
```diff
@@ -28,56 +28,56 @@ services:
         ipv4_address: 172.21.0.3
   compactor-0:
     extends:
-      file: ../../docker/docker-compose.yml
+      file: ../../docker/docker-compose-minio.yml
```
[the remark this answers is not in the document]
Comment (huangjw806) [reply integration_tests/doris-sink/docker-compose.yml:31]: As mentioned above, I am not sure how to run it using standalone for doris-sink and feature-store.
https://github.com/risingwavelabs/risingwave/pull/13814#issuecomment-1840461239

**Comment by huangjw806** [comment]
> "Doris Sink" is because we need to ensure that the entire cluster adopts the same set of network IP, otherwise it will result in RW being unable to access Doris. Is there any other way to solve the whole problem? The Feature Store needs to transfer UDF files to Docker, and I'm not sure if there is any other solution.

These two cases are not in the daily test list. We can temporarily not enable standalone mode in this PR. Maybe you can try to see if you can run them in standalone mode?

**Comment by huangjw806** [comment]
> Did you run it using the nightly version? I ran on the nightly last time. Please use the cluster mode if it just cannot run.

I use ‘nightly-20231205’ version
image

**Review by StrikeW — APPROVED** [review]

**Pull request merged**